packages/data-service/src/instance-detail-helper.ts

```diff
@@ -167,12 +167,15 @@ export function extractPrivilegesByDatabaseAndCollection(
 ): DatabaseCollectionPrivileges {
   const privileges = authenticatedUserPrivileges ?? [];
 
-  const filteredPrivileges =
+  const filteredPrivileges = (
     requiredActions && requiredActions.length > 0
       ? privileges.filter(({ actions }) => {
           return requiredActions.every((action) => actions.includes(action));
         })
-      : privileges;
+      : privileges
+  ).filter(({ resource }) => {
+    return resource.db;
```

Comment:
Is this for a case where db is somehow completely undefined (not `""` but like completely missing)? Sould we be a bit more strict with the check here in case we ever need privileges that are applied to all databases or alternatively document with a comment that we filtering out both all databases privileges and privileges that don't have a database here so we don't forget next time we do changes in this area?

Reply:
Yeah you are right, this PR was a super quick fix attempt for a bug, but is too late and i gave up. We can fix it tomorrow where it should be fixed i guess.

The bug is this one:

<img width="381" alt="Screenshot 2021-11-11 at 21 33 20" src="https://user-images.githubusercontent.com/334881/141380539-a3857b13-5f7c-4720-95ae-150167ac4937.png">
 
Happens with the `sharded` docker setup. This fix works, but probably we want to just ignore empty databases somewhere else, or understand why the guard that ignored them didn't work.

Reply:
Did some digging in the docs today and found a [nice and thorough](https://docs.mongodb.com/manual/reference/resource-document/#std-label-resource-document) doc about all possible roles, _but then_ I also found that there are some undocumented ones like `system_buckets` 😆 So it seems like as much as it would be nice to avoid, realistically the only viable option we have is indeed to check if `database` and `collection` are `undefined` to avoid running into something similar in the future in case some other resource will be quietly added. I updated the code a little to clarify what exactly we are filtering out there and added some tests to document the behavior, I hope you don't mind